antoniovmonge/django-rest-prod
# Request trip

# **Title**: Implement Driver Broadcast Messaging Upon Login

## Description:

This PR introduces the core functionality for broadcasting messages to active drivers.

## Key changes:

- Driver Pool: Establishes a mechanism to add drivers to an active pool upon login.
- Broadcast System: Implements a backend process to send messages to all drivers in the pool.
- Real-time Updates: Leverages WebSockets to deliver messages to drivers instantly.
- Test Scenario: Includes a test case to:
    1. Create a new driver account.
    2. Simulate driver login.
    3. Trigger a broadcast message.
    4. Verify the driver receives the message in the app.

## Impact:

Enhanced communication: Enables real-time updates for drivers, leading to improved user experience and efficient dispatching.
Foundation for future features: Provides the infrastructure for additional broadcast features like promotions, alerts, and location-based messages.
